kong/api/routes/certificates.lua

```diff
@@ -1,317 +1,80 @@
-        create_certificate(self, dao_factory, helpers)
-        return -- avoid tail call
-      end
-
-      -- id present in body
-      update_certificate(self, dao_factory, helpers)
+      return helpers.responses.send_HTTP_CREATED(data)
     end,
   },
 
+  ["/certificates/:certificates"] = {
+    before = get_cert_by_sni_or_id,
 
-  ["/certificates/:sni_or_uuid"] = {
-    before = function(self, dao_factory, helpers)
-      if utils.is_valid_uuid(self.params.sni_or_uuid) then
-        self.ssl_certificate_id = self.params.sni_or_uuid
-
-      else
-        -- get requested SNI
-
-        local row, err = dao_factory.ssl_servers_names:find {
-          name = self.params.sni_or_uuid
-        }
-        if err then
-          return helpers.yield_error(err)
-        end
-
-        if not row then
-          return helpers.responses.send_HTTP_NOT_FOUND()
-        end
+    -- override to include the snis list when getting an individual certificate
+    GET = function(self, db, helpers)
+      local pk = { id = self.params.certificates }
 
-        -- cache certificate row id
-
-        self.ssl_certificate_id = row.ssl_certificate_id
+      local cert, _, err_t = db.certificates:select_with_name_list(pk)
+      if err_t then
+        return endpoints.handle_error(err_t)
       end
 
-      self.params.sni_or_uuid = nil
+      return helpers.responses.send_HTTP_OK(cert)
     end,
 
-
-    GET = function(self, dao_factory, helpers)
-      local row, err = dao_factory.ssl_certificates:find {
-        id = self.ssl_certificate_id
-      }
-      if err then
-        return helpers.yield_error(err)
+    -- override to accept the snis param when updating a certificate
+    PATCH = function(self, db, helpers)
+      local pk = { id = self.params.certificates }
+      local cert, _, err_t = db.certificates:update_with_name_list(pk, self.args.post)
+      if err_t then
+        return endpoints.handle_error(err_t)
       end
-
-      if not row then
-        return helpers.responses.send_HTTP_NOT_FOUND()
-      end
-
-      -- add list of other SNIs for this certificate
-
-      row.snis = setmetatable({}, cjson.empty_array_mt)
-
-      local rows, err = dao_factory.ssl_servers_names:find_all {
-        ssl_certificate_id = self.ssl_certificate_id
-      }
-      if err then
-        return helpers.yield_error(err)
-      end
-
-      for i = 1, #rows do
-        row.snis[i] = rows[i].name
-      end
-
-      return helpers.responses.send_HTTP_OK(row)
-    end,
-
-
-    PATCH = function(self, dao_factory, helpers)
-      self.params.id = self.ssl_certificate_id
-      update_certificate(self, dao_factory, helpers)
+      return helpers.responses.send_HTTP_OK(cert)
     end,
+  },
 
-
-    DELETE = function(self, dao_factory, helpers)
-      return crud.delete({
-        id = self.ssl_certificate_id
-      }, dao_factory.ssl_certificates)
-    end,
-  }
+  ["/certificates/:certificates/snis"] = {
+    before = get_cert_by_sni_or_id,
+  },
```

Comment:
You can do `GET` and `POST` on it.
```
GET /certificates/xxx/snis
POST /certificates/xxx/snis { name: "potato.com" }
```

`GET` lists the "real sni entities" associated with the certificate, not "magical strings". It's the only place where you can get all the ids of the snis associated to a cert via the admin api.

`POST` allows adding a single sni to the existing list of snis (so if a cert has 100 snis, you can use this instead of doing a PATCH on the cert and passing 101 snis as an array).